README.md

```diff
@@ -51,28 +43,34 @@ if (scope != null) {
 
 ### Starting a new Span
 
-The common case starts an `Scope` that's automatically registered for intra-process propagation via `ScopeManager`. The best practice is to use a try-with-resources pattern which handles Exceptions and early returns:
+The common case starts an `Scope` that's automatically registered for intra-process propagation via `ScopeManager`.
+Note that the default behaviour of `startActive()` is not to finish the span on `Scope.close()`.
+This was made on purpose because `try-with-resources` construct would finished the span before
+`catch` or `finally` blocks are executed, which makes logging exceptions impossible.
+This behaviour can be overridden by `startActive(finishOnClose)` which is demonstrated later in this section.
 
 ```java
 io.opentracing.Tracer tracer = ...;
 ...
-try (Scope scope = tracer.buildSpan("someWork").startActive()) {
+Span span = tracer.buildSpan("someWork").startManual();
+try (Scope scope = tracer.scopeManager().activate(span))
     // Do things.
-    //
-    // If we create async work, `scope.span()` allows us to pass the `Span` along as well.
+} catch {
+    Tags.ERROR.set(span(), true);
+} finally {
+    span.finish();
 }
 ```
 
-The above is semantically equivalent to the more explicit try-finally version:
+The following code uses `try-with-resources` to finish the span.
 
 ```java
 io.opentracing.Tracer tracer = ...;
 ...
-Scope scope = tracer.buildSpan("someWork").startActive();
-try {
+try (Scope scope = tracer.buildSpan("someWork").startActive(true)) {
     // Do things.
-} finally {
-    scope.deactivate();
+    //
+    // If we create async work, `scope.span()` allows us to pass the `Span` along as well.
```

Comment:
This seems misleading.  If async work is spawned here the span will be finished by the time async thread has a chance to start.

Reply:
It was in the original text. I will rephrase it.